lib/policyEvaluator/utils/conditions.js

```diff
@@ -0,0 +1,407 @@
+    map.set('s3:VersionId', requestContext.headers['x-amz-version-id']);
+    // s3:LocationConstraint -- Used to restrict creation of bucket
+    // in certain region.  Only applicable for CreateBucket
+    map.set('s3:LocationConstraint', requestContext.locationConstraint);
+    // s3:delimiter is delimiter for listing request
+    map.set('s3:delimiter', requestContext.query.delimiter);
+    // s3:max-keys is max-keys for listing request
+    map.set('s3:max-keys', requestContext.query['max-keys']);
+    // s3:prefix is prefix for listing request
+    map.set('s3:prefix', requestContext.query.prefix);
+    // s3 auth v4 additional condition keys
+    // (See http://docs.aws.amazon.com/AmazonS3/latest/API/
+    // bucket-policy-s3-sigv4-conditions.html)
+    // s3:signatureversion -- Either "AWS" for v2 or
+    // "AWS4-HMAC-SHA256" for v4
+    map.set('s3:signatureversion', requestContext.signatureVersion);
+    // s3:authType -- Method of authentication: either "REST-HEADER",
+    // "REST-QUERY-STRING" or "POST"
+    map.set('s3:authType', requestContext.authType);
+    // s3:signatureAge is the length of time, in milliseconds,
+    // that a signature is valid in an authenticated request.  So,
+    // can use this to limit the age to less than 7 days
+    map.set('s3:signatureAge', requestContext.signatureAge);
+    // s3:x-amz-content-sha256 - Valid value is "UNSIGNED-PAYLOAD"
+    // so can use this in a deny policy to deny any requests that do not
+    // have a signed payload
+    map.set('s3:x-amz-content-sha256',
+        requestContext.headers['x-amz-content-sha256']);
+    return map.get(key);
+};
+
+
+// Wildcards are allowed in certain string comparison and arn comparisons
+// Permitted in StringLike, StringNotLike, ArnLike and ArnNotLike
+// This restriction almost matches up with where variables can be used in
+// conditions so converting ${*}, ${?} and ${$} as part of the wildcard
+// transformation instead of the variable substitution works
+// (except for the StringEquals, StringNotEquals, ArnEquals and
+// ArnNotEquals conditions where wildcards
+// not allowed but variables are allowed).  For those 4 operators, we switch
+// out ${*}, ${?} and ${$} in the convertConditionOperator function.
+function convertSpecialChars(string) {
+    function characterMap(char) {
+        const map = {
+            '${*}': '*',
+            '${?}': '?',
+            '${$}': '$',
+        };
+        return map[char];
+    }
+    return string.replace(/(\$\{\*\})|(\$\{\?\})|(\$\{\$\})/g,
+    characterMap);
+}
+
+/**
+ * convertToEpochTime checks whether epoch or ISO time and converts to epoch
+ * if necessary
+ * @param {string | array} time - value or values to be converted
+ * @return {string | array} converted value or values
+ */
+function convertToEpochTime(time) {
+    function convertSingle(item) {
+        // If ISO time
+        if (item.indexOf(':') > -1) {
+            return new Date(item).valueOf();
```

Comment:
This line would return a number and not a string. To be consistent, we should always return number/array of numbers since this function is used along with `NumericEquals` and others. So, if it's not ISO and its Epoch it needs to be converted to a number here as well.